devito/types/equation.py

```diff
@@ -255,3 +254,17 @@ def solve(eq, target, **kwargs):
         return target.new_from_mat(sols)
     else:
         return sols[0]
+
+
+def eval_t(expr):
+    """
+    Evaluate all time derivatives in the expression
+    """
+    try:
+        assert any(d.is_Time for d in expr.dims)
+        return expr.evaluate
+    except:
+        try:
```

Comment:
nested exceptions just be avoided. This is nitpicking (only two levels), but the more pythonic way is to `pass`, de-indent the nested `try-except`, add a comment, and go